src/app/ui/admin-page/admin-page.component.ts

```diff
@@ -57,17 +60,43 @@ export class AdminPageComponent implements OnInit {
   }
 
   openSeasonDialog(user: User) {
-    const addSeasonDialog = this.dialog.open(AddSeasonDialogComponent, {width: '400px'});
-    addSeasonDialog.afterClosed().subscribe( data => {
+    const addSeasonDialog = this.dialog.open(AddSeasonDialogComponent, { width: '400px' });
+    addSeasonDialog.afterClosed().subscribe(data => {
       if (data) {
         this.seasonService.createSeason(data.name, data.startDate, data.endDate, {
           uid: user.uid,
           displayName: user.displayName
         })
-        .catch((err) => {
-          console.log(err);
-          this.notifyService.update(`Something went wrong. Please try again later.`, 'error');
-        });
+          .catch((err) => {
+            console.log(err);
+            this.notifyService.update(`Something went wrong. Please try again later.`, 'error');
+          });
+      }
+    });
+  }
+
+  openBadgeDialog(user) {
+    const addBadgeDialog = this.dialog.open(AddBadgeDialogComponent,
+      {
+        data: {
+          user
+        },
```

Comment:
Only need uid so no need to pass the whole object
```suggestion
        data: user.uid
```